pexec/managed_process.go

```diff
@@ -173,12 +175,13 @@ func (p *managedProcess) Start(ctx context.Context) error {
 	return nil
 }
 
-// manage is the watchdog of the process. Any time it detects
-// the process has ended unexpectedly, it will restart it. It's
-// possible and okay for a restart to be in progress while a Stop
-// is happening. As a means simplifying implementation, a restart
-// spawns new goroutines by calling Start again and lets the original
-// goroutine die off.
+// manage is the watchdog of the process. If the process has ended
+// unexpectedly, onCrashHandler will be called. If onCrashHandler is unset or
+// returns true, manage will restart the process. Note that onCrashHandler may
+// be called multiple times if it returns true. It's possible and okay for a
```

Comment:
> Not that onCrashHandler may be called multiple times if it returns true.

I'm not sure how I feel about this behavior. At first, I thought we would only call the `onCrashHandler` once when processes crash. If the process crashes, though, `manage` will restart both the process _and_ another `manage` goroutine; this leads to potentially infinite restarts, and I can't think of a way to differentiate between sequential quick crashes where `onCrashHandler` should only be called once and long runs of the process with intermittent crashing where `onCrashHandler` should maybe be called on each crash.